### PR TITLE
Selenoid mesos

### PR DESCRIPTION
**Pull request by alisa0301**
add mesos-framework implementation
add implementation of interface Starter
add flags: -mesos and -zk

### CODE REVIEW

**Comment by vania-pooh** [comment]
We confirm that we have seen this PR. Give us some time to review.

File: service/service.go [hunk 1]
```diff
@@ -73,21 +75,32 @@ func (m *DefaultManager) Find(caps session.Caps, requestId uint64) (Starter, boo
...
 		return nil, false
 	}
-	switch service.Image.(type) {
-	case string:
-		if m.Client == nil {
-			return nil, false
-		}
-		log.Printf("[%d] [USING_DOCKER] [%s] [%s]", requestId, browserName, version)
-		return &Docker{
+	if m.Environment.MesosMasterUrl != "" || m.Environment.Zookeeper != "" {
```
Comment (aandryashin) [inline service/service.go]: please move mesos section under switch to default case

Comment (aandryashin) [inline service/service.go]: please move mesos section to switch under string case and there choose to use docker or mesos

File: mesos/scheduler/messages.go [hunk 2]
```diff
@@ -0,0 +1,332 @@
+package scheduler
+
+//Универсальная структура для хранения  ID
```
Comment (vania-pooh) [inline mesos/scheduler/messages.go]: Please avoid russian in code. Should be readable by anyone in community.

File: mesos/scheduler/messages.go [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+}
+
+type Command struct {
+	Env   Env  `json:"environment"`
+	Shell bool `json:"shell"`
+}
+
+type TaskInfo struct {
+	Name      string     `json:"name"`
+	TaskID    ID         `json:"task_id"`
```
Comment (vania-pooh) [inline mesos/scheduler/messages.go]: We prefer CamelCase instead of underscores.

File: start.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
```
Comment (vania-pooh) [inline start.sh]: This one should be removed or moved to docs.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
+version: "3.6"
```
Comment (vania-pooh) [inline docker-compose.yml]: This file is useless. Should be removed or moved to docs.

**Comment by vania-pooh** [comment]
We unfortunately decided to not merge this. Apache Mesos market share is constantly decreasing. Also this implementation would be too much complicated for us to maintain.

**Pull request closed without merging**